encoding/m3tsz/decoder.go

```diff
@@ -27,18 +27,19 @@ import (
 )
 
 type decoder struct {
-	opts Options
+	opts         encoding.Options
+	intOptimized bool
 }
 
 // NewDecoder creates a decoder.
-func NewDecoder(opts Options) encoding.Decoder {
+func NewDecoder(opts encoding.Options, intOptimized bool) encoding.Decoder {
```

Comment:
Nit: can we make `opts` be the second parameter here?  This is traditionally the pattern we have with other components that take `opts` as well as some parameters in their constructors.